### DISCUSSION TITLE
TScrollBar's fly away, when an other TWindow becomes the focus.

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#128] [Q&A] [answered]
this is the normal view (when the blue window have the focus):
![grafik](https://github.com/magiblot/tvision/assets/3682603/d8f3908b-24f9-4c40-886e-87a9b5d65e59)

this is the view, when the green window have the focus.
you can see, that the TScrollBar's are missing ?
![grafik](https://github.com/magiblot/tvision/assets/3682603/14882f72-09eb-409c-9778-385205ea0229)

And this is the view on focus the blue window.
See the TScrollBar at right side of the TListBox - it will be change when I resize the window:
![grafik](https://github.com/magiblot/tvision/assets/3682603/1be6ae17-5609-42da-af09-79e2f65d9623)

The TMyListBox, which I suib class (inherited from TListBox) will be stay on the place (it moves with resize of the blue window) but it does not change the size itself. So, when the blue window becomes a minimum, the TMyListBox will be override the bottom Frame. But it is okay - it work fine.

Thanks for reading, and watching
paule32

**Comment by magiblot** [comment, marked as answer]
Hi Jens,

The disappearing scrollbars are a design feature of Turbo Vision. In your case, the logic that is being applied is in `TEditor::setState` and `TListViewer::setState`. These methods hide or show the scrollbars on purpose.

In order to change this behaviour, you should override these methods in your own subclass. I believe the simplest solution may be to always call the `show()` method on the scrollbars, like this:

```c++
void TMyEditor::setState(ushort aState, bool enable)
{
    TEditor::setState(aState, enable);
    if (hScrollBar)
        hScrollBar->show();
    if (vScrollBar)
        vScrollBar->show();
}

void TMyListBox::setState(ushort aState, bool enable)
{
    TListBox::setState(aState, enable);
    if (hScrollBar)
        hScrollBar->show();
    if (vScrollBar)
        vScrollBar->show();
}
```

When it comes to the size of the list box, the issue is again the property `growMode`. In this case you should set it to `gfGrowHiX | gfGrowHiY`.

**Reply by paule32** [reply]
This is my ListBox code:
```
class TMyListBox : public TListBox {
public:
    TMyListBox(
        const TRect& listBounds,
        int   cols ,
        TScrollBar * listScrollBar):
        TListBox( listBounds,  cols, listScrollBar) {

        growMode = gfGrowHiX | gfGrowHiY;
    }
};
```
when I use this, the Listbox will overwrite the rest of the window:
![grafik](https://github.com/magiblot/tvision/assets/3682603/1d99f58f-6454-4174-9814-4ee013d6b650)

**Reply by paule32** [reply]
okay, I have done it with this Code:
```
class TMyListBox : public TListBox {
public:
    TMyListBox(
        const TRect& listBounds,
        int   cols ,
        TScrollBar * listScrollBar):
        TListBox( listBounds,  cols, listScrollBar) {

        growMode = gfGrowHiX | gfGrowHiY | gfGrowLoX;
    }
};
```
